src/components/popular.js

```diff
@@ -9,28 +20,29 @@ class Popular extends React.Component {
 		this.updateLanguage = this.updateLanguage.bind(this);
 	}
 
-	updateLanguage(lang) {
+	updateLanguage(synth, event) {
```

Comment:
You'll notice the `args` passed to this function have changed a bit. This has to do with the scope of the function. 

Take a look at this example: 

```
var parent = function () {
  const someVariable = 'hello';
  return function closureFunction () {
    const anotherVariable = 'world';
    console.log(someVariable);
  }
  console.log(anotherVariable);
}
```
and you were to run both of these functions: `parent()()`
You would get and output of:
```
'hello'
undefined
```
The first console.log with `someVariable` prints in the function that inside the other function, but the console.log is `undefined` for `anotherVariable` because the scope of that second variable is limited to the child function.

`this` is a reserved keyword variable that stores the scope of any function or object. In the context of the example above, the function `closureFunction` has scope of the `parent` function and `this` will be the scope of the `parent` function. 

There are 3 functions that allow you change the scope of a function: apply, call, and bind. If I changed the example above to 

```
var qwer = function () {
  const someVariable = 'hello';
  return function closureFunction () {
    const anotherVariable = 'world';
    console.log(this);
  }.bind('I will be stored in this!')
}
```
The console.log will print out 'I will be stored in this!' even though the parent function is in the same scope. We've changed it using the `bind` function. So let's apply this to the react example you asked.  

In React, there are many levels of functions in the component Object. When you tell an event function, link `onClick` to call a function you have defined, the scope of where that function is being called is not shared with the function you've created: `updateLanguage`. In the constructor of the application, which is run right as all of this is created, we `bind` (or set scope of that function to the scope of the component) so now when the click event is fired, it passes the proper scope to the `updateLangauge` function. 

There is a LOT more to discuss and many more examples we could use to make this more clear. But this is what you're running up against.